core/invoker/src/main/scala/org/apache/openwhisk/core/containerpool/ContainerPool.scala

```diff
@@ -112,7 +112,17 @@ class ContainerPool(childFactory: ActorRefFactory => ActorRef,
             // Schedule a job to a warm container
             ContainerPool
               .schedule(r.action, r.msg.user.namespace.name, freePool)
-              .map(container => (container, "warm"))
+              .map { container =>
+                val warmState = container._2 match {
```

Comment:
This cannot be done exactly, since the logged string indicating state does not map directly to specific `ContainerData` types; this is one of several problems where `ContainerData` is reused as a) indicator of container status in `ContainerPool` and b) state management within `ContainerProxy`. There is some overlap, but there are a few unique concerns for each so there is some mismatch when used in one or the other. I updated ContainerData to have `initingState`, but it cannot be used in all cases.